src/Microsoft.VisualStudio.Composition/LazyMetadataWrapper.cs

```diff
@@ -195,6 +197,32 @@ public bool Remove(KeyValuePair<string, object> item)
             return metadata;
         }
 
+        internal static bool TryGetMetadataValueTypeRef(IReadOnlyDictionary<string, object> metadata, string key, Resolver resolver, out object typeRef)
```

Comment:
>object [](start = 137, length = 6)

Why is `typeRef` typed as `object` here? If it's always a `TypeRef`, why not type it as such? Or if it might not be one, why name it that way?
From the assignment below, it looks like in fact this value might be set to whatever metadata value the metadata is set to, including `Type` itself.
At a minimum, let's please add an xml doc comment explaining what this method and all its parameters and return values signify.